### PR TITLE
Lion IMBA

### PR DESCRIPTION
**Pull request by Shushishtok**
Tooltips, Lion.lua, npc_ability_custom updated to include Lion IMBA

### CODE REVIEW

File: game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+	local extra_spike_AOE = ability:GetLevelSpecialValueFor("extra_spike_AOE", ability_level)
+	local wait_interval = ability:GetLevelSpecialValueFor("wait_interval", ability_level)
+			
+	-- Spikes ignore target if it was already hit this cast
+	if target.hit_by_earth_spikes then
+		return nil
+	end
+	
+	-- Mark target as hit by Earth Spikes
+	target.hit_by_earth_spikes = true
```
Comment (brandonsturgeon) [inline game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua]: I don't make dota addons at all, so please excuse my question, it's just based off curiosity;

If `target` is just a point on the ground, what happens when you set `hit_by_earth_spikes = true` on it?

Comment (Shushishtok) [inline game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua]: Hey there,
This function is called each time an enemy is hit by earth spikes. This means that "target" is always an enemy unit/hero in this case.

The cast function uses the target point to know in which direction the spikes should go, if it's an enemy target I get his location to do the same.

Comment (brandonsturgeon) [inline game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua]: Ahhh, I see :) I didn't realize these were spread across different functions.

Thanks!

I'm sure you're already all over it, but flame amphibian suggests that you post your IMBA suggestion on the current IMBA suggestion thread on Reddit before making a pull request.

**Comment by Shushishtok** [comment]
This is a concept that he asked for. I've built IMBA Lion with his instructions of how the skills should work and interact.

The Skywrath Mage concept was made by me entirely though, as a test for myself whether I could develop a hero from scratch.

Shush



On Dec 8, 2016 9:04 PM, Brandon Sturgeon <notifications@github.com> wrote:<br type="attributi

@brandonsturgeon commented on this pull request.

________________________________

In game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua<https://github.com/fcalife/dota_imba/pull/78>:

> +     local knock_up_time = ability:GetLevelSpecialValueFor("knock_up_time", ability_level)
+       local damage = ability:GetLevelSpecialValueFor("damage", ability_level)
+       local stun_duration = ability:GetLevelSpecialValueFor("stun_duration", ability_level)
+       local spike_speed = ability:GetLevelSpecialValueFor("spike_speed", ability_level)
+       local travel_distance = ability:GetLevelSpecialValueFor("travel_distance", ability_level)
+       local spikes_radius = ability:GetLevelSpecialValueFor("spikes_radius", ability_level)
+       local extra_spike_AOE = ability:GetLevelSpecialValueFor("extra_spike_AOE", ability_level)
+       local wait_interval = ability:GetLevelSpecialValueFor("wait_interval", ability_level)
+
+       -- Spikes ignore target if it was already hit this cast
+       if target.hit_by_earth_spikes then
+               return nil
+       end
+
+       -- Mark target as hit by Earth Spikes
+       target.hit_by_earth_spikes = true


Ahhh, I see :) I didn't realize these were spread across different functions.

Thanks!

I'm sure you're already all over it, but flame amphibian suggests that you post your IMBA suggestion on the current IMBA suggestion thread on Reddit before making a pull request.

-

File: game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua (hunk 1 above)
Comment (Shushishtok) [inline game/dota_addons/dota_imba/scripts/vscripts/hero/hero_lion.lua]: This is a concept that he asked for. I've built IMBA Lion with his instructions of how the skills should work and interact.

The Skywrath Mage concept was made by me entirely though, as a test for myself whether I could develop a hero from scratch.

Shush

**Comment by fcalife** [comment]
again, some weird conflict on addon_english.txt, but I'll be adding this manually.

Thank you!

**Pull request closed without merging**